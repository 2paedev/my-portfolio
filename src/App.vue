<template>
  <div class="app">
    <section class="app__block--no-margin">
      <personal-info></personal-info>
    </section>

    <section
      class="app__block"
      data-aos="fade-right"
      data-aos-offset="200"
      data-aos-delay="150"
      data-aos-duration="1300"
    >
      <skills></skills>
    </section>

    <section
      class="app__block"
      data-aos="fade-left"
      data-aos-offset="200"
      data-aos-delay="150"
      data-aos-duration="1300"
    >
      <projects></projects>
    </section>

    <section
      class="app__block"
      data-aos="slide-up"
      data-aos-offset="300"
      data-aos-delay="100"
      data-aos-duration="1000"
    >
      <hobbies></hobbies>
    </section>
  </div>
</template>

<script>
import PersonalInfo from './components/PersonalInfo.vue';
import Skills from './components/Skills.vue';
import Projects from './components/Projects.vue';
import Hobbies from './components/Hobbies.vue';

export default {
  name: `App`,
  components: {
    PersonalInfo,
    Skills,
    Projects,
    Hobbies
  }
};
</script>

<style lang="scss" scoped>
.app {
  &__block {
    margin: 15rem;
    &--no-margin {
      margin: 0;
    }
  }
}
</style>
