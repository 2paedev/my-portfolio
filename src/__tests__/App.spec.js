import { shallowMount } from '@vue/test-utils';
import App from '../App.vue';

describe('App.vue', () => {
  it('renders the same HTML', () => {
    const wrapper = shallowMount(App);
    expect(wrapper.element).toMatchSnapshot();
  });

  it('render PersonalInfo component', () => {
    const wrapper = shallowMount(App);
    const component = wrapper.find({ name: 'personal-info' });
    expect(component.exists()).toBeTruthy();
  });

  it('render Skills component', () => {
    const wrapper = shallowMount(App);
    const component = wrapper.find({ name: 'skills' });
    expect(component.exists()).toBeTruthy();
  });

  it('render Projects component', () => {
    const wrapper = shallowMount(App);
    const component = wrapper.find({ name: 'projects' });
    expect(component.exists()).toBeTruthy();
  });

  it('render Hobbies component', () => {
    const wrapper = shallowMount(App);
    const component = wrapper.find({ name: 'hobbies' });
    expect(component.exists()).toBeTruthy();
  });
});
