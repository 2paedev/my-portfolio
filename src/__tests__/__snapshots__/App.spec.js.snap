// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App.vue renders the same HTML 1`] = `
<div
  class="app"
>
  <section
    class="app__block"
    data-aos="zoom-out"
    data-aos-delay="100"
    data-aos-duration="3000"
  >
    <personal-info-stub />
  </section>
   
  <section
    class="app__block"
    data-aos="fade-right"
    data-aos-delay="150"
    data-aos-duration="1000"
    data-aos-offset="500"
  >
    <skills-stub />
  </section>
   
  <section
    class="app__block"
    data-aos="flip-left"
    data-aos-delay="100"
    data-aos-duration="2500"
    data-aos-offset="350"
  >
    <projects-stub />
  </section>
   
  <section
    class="app__block"
    data-aos="slide-up"
    data-aos-delay="100"
    data-aos-duration="1000"
    data-aos-offset="300"
  >
    <hobbies-stub />
  </section>
</div>
`;
